lib/remote_command_runner.rb

```diff
@@ -56,4 +56,86 @@ def run(*args)
   rescue Cheetah::ExecutionFailed => e
     raise ExecutionFailed.new(e)
   end
+
+  # Copy a local file to the remote system.
+  #
+  # +source+:: Path to the local file
+  # +destination+:: Path to the remote file or directory. If +destination+ is a
+  #                 path, the same filename as +source+ will be used.
+  # +opts+:: Options to modify the attributes of the remote file.
+  #
+  #          Available options:
+  #          [owner]:: Owner of the file, e.g. "tux"
+  #          [group]:: Group of the file, e.g. "users"
+  #          [mode]:: Mode of the file, e.g. "600"
+  def inject_file(source, destination, opts = {})
+    # Append filename (taken from +source+) to destination if it is a path, so
+    # that +destination+ is always the full target path including the filename.
+    destination += File.basename(source) if destination.end_with?("/")
+
+    Cheetah.run(
+      "scp",
+      "-o",
+      "UserKnownHostsFile=/dev/null",
+      "-o",
+      "StrictHostKeyChecking=no",
+      source,
+      "root@#{@ip}:#{destination}"
+    )
+
+    if opts[:owner] || opts[:group]
+      owner_group = opts[:owner] || ""
+      owner_group += ":#{opts[:group]}" if opts[:group]
+      run "chown", "-R", owner_group, destination
+    end
+
+    if opts[:mode]
+      run "chmod", opts[:mode], destination
+    end
+  rescue Cheetah::ExecutionFailed => e
+    raise ExecutionFailed.new(e)
+  end
+
+  def extract_file(source, destination)
+    Cheetah.run(
+      "scp",
+      "-o",
+      "UserKnownHostsFile=/dev/null",
+      "-o",
+      "StrictHostKeyChecking=no",
+      "root@#{@ip}:#{source}",
+      destination
+    )
+  rescue Cheetah::ExecutionFailed => e
+    raise ExecutionFailed.new(e)
+  end
+
+  def inject_directory(source, destination, opts = {})
```

Comment:
Assignment Branch Condition size for inject_directory is too high. [15.13/15]<br>Perceived complexity for inject_directory is too high. [8/7]